### PR TITLE
Add integration tests for the new --eus option

### PR DESCRIPTION
**Pull request by danmyway**
* add new test scenarios
* add the --eus option to the test validating a fallback to the regular repositories when using non-eus pool

Jira Issues: [RHELC-880](https://issues.redhat.com/browse/RHELC-880)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    _restore_eus_mapping()
+
+
+REGULAR_REPOID_MSG = "RHEL repository IDs to enable: rhel-8-for-x86_64-baseos-rpms, rhel-8-for-x86_64-appstream-rpms"
+EUS_REPOID_MSG = (
+    "RHEL repository IDs to enable: rhel-8-for-x86_64-baseos-eus-rpms, rhel-8-for-x86_64-appstream-eus-rpms"
+)
+
+EUS_SYS_PRE_EUS_PHASE = '"8.8": "2042-04-08",'
```
Comment (r0x0d) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:40]: 2042 is just around the corner

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+EUS_SYS_EUS_PHASE = '"8.8": "1970-01-01",'
+
+
+eus_support_parameters = [
+    # System not recognized as an EUS candidate with --eus option used
+    ("--eus", REGULAR_REPOID_MSG, "", False, False),
+    # System recognized as and EUS candidate before the start of EUS phase without --eus option
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:47]:
```suggestion
    # System recognized as an EUS candidate before the start of EUS phase without --eus option
```

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    ("", REGULAR_REPOID_MSG, EUS_SYS_PRE_EUS_PHASE, False, False),
+    # System recognized as and EUS candidate before the start of EUS phase with --eus option
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:49]:
```suggestion
    # System recognized as an EUS candidate before the start of EUS phase with --eus option
```

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    ("--eus", EUS_REPOID_MSG, EUS_SYS_PRE_EUS_PHASE, True, False),
+    # System recognized as and EUS candidate after the start of EUS phase without --eus option
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:51]:
```suggestion
    # System recognized as an EUS candidate after the start of EUS phase without --eus option
```

File: tests/integration/tier0/non-destructive/eus/main.fmf [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        The report is expected to print out a WARNING and advise to use the --eus option.
+
+tag+:
+    - eus
+
+/eus_support:
+    enabled: false
+    adjust+:
+        enabled: true
+        when: distro == alma-8-latest, rocky-8-latest
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/main.fmf:26]: We will need to change this to alma-8.8 and rocky-8.8 once they are not latest anymore. Maybe add and `todo` or `fixme`  with a mention about that

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    with convert2rhel(
+        "analyze -y --debug --no-rpm-va --serverurl {} -u {} -p {} {}".format(
+            env.str("RHSM_SERVER_URL"), env.str("RHSM_USERNAME"), env.str("RHSM_PASSWORD"), additional_options
+        )
+    ) as c2r:
+        # If the system is an EUS candidate and current date is past the beginning of the EUS phase
+        # we set the eus_recommend parameter to True
+        # We expect the report to print out a warning to use --eus option
+        if eus_recommend:
+            c2r.expect(repoid_message, timeout=120)
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:91]: this line can be placed before the `if` as it is also in the `else` section

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+            c2r.expect("EUS_SYSTEM_CHECK::EUS_COMMAND_LINE_OPTION_UNUSED")
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:92]:
```suggestion
            c2r.expect("Pre-conversion analysis report")
            c2r.expect("EUS_SYSTEM_CHECK::EUS_COMMAND_LINE_OPTION_UNUSED")
```

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        else:
+            c2r.expect(repoid_message, timeout=120)
+            # In case the EUS repositories should get enabled, we set is_eus parameter to True
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/test_eus_support.py:95]: I would remove this comment as it is not giving any different information then the code itself.
Or rephrase it to:
```suggestion
            # If  is_eus is True, expect a corresponding EUS message displayed
```

File: tests/integration/tier0/non-destructive/eus/main.fmf [hunk 9]
```diff
@@ -0,0 +1,30 @@
+summary: |
+    Verify the EUS releases are treated correctly
+description: |
+    Test verifying correct behavior when converting EUS candidates.
+    EUS_MINOR_VERISONS mapping in convert2rhel/systeminfo.py is modified to mock the different scenarios.
+    Verified scenarios (handled by patest parametrization):
+    1/ The running minor version is removed from the mapping to not be considered an EUS candidate.
+        The --eus option is used in the command and enabling regular (non-eus) repoids is verified.
+    2/ The date of the start of the EUS phase is set far to the future to simulate the EUS phase
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/main.fmf:9]:
```suggestion
    2/ The start date of the EUS phase is set far to the future to simulate the EUS phase
```

File: tests/integration/tier0/non-destructive/eus/main.fmf [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        did not start yet.
+        The --eus option *is not used* and regular repoids enablement is expected.
+    3/ The date of the start of the EUS phase is set far to the future to simulate the EUS phase
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/main.fmf:12]:
```suggestion
    3/ The start date of the EUS phase is set far to the future to simulate the EUS phase
```

File: tests/integration/tier0/non-destructive/eus/main.fmf [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        did not start yet.
+        The --eus option *is used* and EUS repoids enablement is expected.
+    4/ The date of the start of the EUS phase is set far to the past to simulate the EUS phase began.
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/main.fmf:15]:
```suggestion
    4/ The start date of the EUS phase is set far to the past to simulate the EUS phase began.
```

File: tests/integration/tier0/non-destructive/eus/main.fmf [hunk 12]
```diff
@@ -0,0 +1,30 @@
+summary: |
+    Verify the EUS releases are treated correctly
+description: |
+    Test verifying correct behavior when converting EUS candidates.
+    EUS_MINOR_VERISONS mapping in convert2rhel/systeminfo.py is modified to mock the different scenarios.
+    Verified scenarios (handled by patest parametrization):
+    1/ The running minor version is removed from the mapping to not be considered an EUS candidate.
+        The --eus option is used in the command and enabling regular (non-eus) repoids is verified.
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/eus/main.fmf:8]:
```suggestion
    1/  The system is considered as a non-EUS. This is done by removing the 8.8 version from the mapping file (note that this workaround works only on 8.8 systems).
        The --eus option is used in the command. Only regular (non-eus) repoids should be enabled.
```

**Review by kokesak — CHANGES_REQUESTED** [review]

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py (hunk 1 above)
Comment (danmyway) [reply tests/integration/tier0/non-destructive/eus/test_eus_support.py:40]: It is comming \o/

File: tests/integration/tier0/non-destructive/eus/main.fmf [hunk 13]
```diff
@@ -0,0 +1,30 @@
+summary: |
+    Verify the EUS releases are treated correctly
+description: |
+    Test verifying correct behavior when converting EUS candidates.
+    EUS_MINOR_VERISONS mapping in convert2rhel/systeminfo.py is modified to mock the different scenarios.
+    Verified scenarios (handled by patest parametrization):
```
[the remark this answers is not in the document]
Comment (danmyway) [reply tests/integration/tier0/non-destructive/eus/main.fmf:6]: Good catch, thank you!

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    ids=["non-eus-sys-eus-opt", "eus-sys-pre-eus-no-opt", "eus-sys-pre-eus-eus-opt", "eus-sys-eus-phase-no-opt"],
+)
+@pytest.mark.test_eus_support
+def test_eus_support(
+    convert2rhel, eus_mapping_update, additional_options, repoid_message, modified_mapping, is_eus, eus_recommend
+):
+    """
+    Test verifying correct behavior when converting EUS candidates.
+    EUS_MINOR_VERISONS mapping in convert2rhel/systeminfo.py is modified to mock the different scenarios.
+    Verified scenarios (handled by patest parametrization):
```
[the remark this answers is not in the document]
Comment (danmyway) [reply tests/integration/tier0/non-destructive/eus/test_eus_support.py:68]: Good catch, thank you!

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py (hunk 6 above)
Comment (danmyway) [reply tests/integration/tier0/non-destructive/eus/test_eus_support.py:91]: You're right, thanks!

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py (hunk 7 above)
Comment (danmyway) [reply tests/integration/tier0/non-destructive/eus/test_eus_support.py:92]: Do we really need to check for that?

File: tests/integration/tier0/non-destructive/eus/main.fmf (hunk 9 above)
Comment (danmyway) [reply tests/integration/tier0/non-destructive/eus/main.fmf:9]: yeah, that sounds better :D

File: tests/integration/tier0/non-destructive/eus/test_eus_support.py (hunk 7 above)
Comment (kokesak) [reply tests/integration/tier0/non-destructive/eus/test_eus_support.py:92]: I guess it is better to check that the message appeared in the report and not somewhere else

**Review by kokesak — APPROVED** [review]
LGTM! Thanks @danmyway

**Comment by danmyway** [comment]
Failures unrelated to the changes, to be fixed in separate PR.
